### PR TITLE
Improvements to Task Search

### PR DESCRIPTION
**Pull request by Calvinp**
Makes the following improvements to Task Search:
- Changes the URL from `/taskSearch` to `/task-search` to be more in line with URLs in the rest of the UI.
- Changes the header from `(Global) Historical Tasks` to `Task Search`.
- Adds labels for the current search parameters so that it's possible to see current search terms (even if the fields have changed).
- Removes the JSON button.
- Changes the log button to be a `file` glyphicon rather than `...`. This is also changed in the Task History table in the request detail page.
- Fixes a bug that caused the table to not update when a new search is performed that returns 0 results.
- Preemptively fixes the bug that occurred in the deployDetail page - contributed to by making api calls in `componentWillMount()`.

### CODE REVIEW

**Comment by Calvinp** [comment]
Did it again... Forgot to cc @tpetr @kwm4385 @wolfd

File: SingularityUI/app/components/taskSearch/TaskSearch.jsx [hunk 1]
```diff
@@ -141,4 +160,18 @@ function mapDispatchToProps(dispatch) {
   };
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(rootComponent(TaskSearch, 'Task Search'));
+let firstLoad = true;
+
+function refresh(props) {
+  if (!firstLoad) {
```
Comment (kwm4385) [inline SingularityUI/app/components/taskSearch/TaskSearch.jsx]: If you only want refresh called when the page is first loaded and not again in an interval, just use the optional `refreshInterval` bool in `rootComponent` to false. `RequestDetail` does this https://github.com/HubSpot/Singularity/blob/decaf/SingularityUI/app/components/requestDetail/RequestDetailPage.jsx#L103

**Comment by kwm4385** [comment]
LGTM

**Comment by wolfd** [comment]
LGTM

**Pull request merged**